.github/JAVASCRIPT_CODESTYLE.md

```diff
@@ -0,0 +1,563 @@
+* [Opening braces go on the same line](#opening-braces-go-on-the-same-line)
+* [Declare one variable per var statement](#declare-one-variable-per-var-statement)
+
+### Naming Conventions
+* [Use lowerCamelCase for variables, properties and function names](#use-lowercamelcase-for-variables-properties-and-function-names)
+* [Use UpperCamelCase for class names](#use-uppercamelcase-for-class-names)
+* [Use UPPERCASE for Constants](#use-uppercase-for-constants)
+
+### Variables
+* [Object / Array creation](#object--array-creation)
+
+### Conditionals
+* [Use the === operator](#use-the--operator)
+* [Use multi-line ternary operator](#use-multi-line-ternary-operator)
+* [Use descriptive conditions](#use-descriptive-conditions)
+
+### Functions
+* [Write small functions](#write-small-functions)
+* [Return early from functions](#return-early-from-functions)
+* [Name your closures](#name-your-closures)
+* [No nested closures](#no-nested-closures)
+* [Method chaining](#method-chaining)
+
+### Comments
+* [Use slashes for comments](#use-slashes-for-comments)
+
+### Miscellaneous
+* [Object.freeze, Object.preventExtensions, Object.seal, with, eval](#objectfreeze-objectpreventextensions-objectseal-with-eval)
+* [Requires At Top](#requires-at-top)
+* [Getters and setters](#getters-and-setters)
+* [Do not extend built-in prototypes](#do-not-extend-built-in-prototypes)
+
+## Formatting
+
+You may want to use [editorconfig.org](http://editorconfig.org/) to enforce the formatting settings in your editor. Use the [MagicMirror² Style Guide .editorconfig file](.editorconfig) to have indentation, newslines and whitespace behavior automatically set to the rules set up below.
+
+### Tabs for indentation
+
+Use tabulator for indenting your code and swear an oath to never mix tabs and
+spaces - a special kind of hell is awaiting you otherwise.
+
+### Newlines
+
+Use UNIX-style newlines (`\n`), and a newline character as the last character
+of a file. Windows-style newlines (`\r\n`) are forbidden inside any repository.
+
+### No trailing whitespace
+
+Just like you brush your teeth after every meal, you clean up any trailing
+whitespace in your JS files before committing. Otherwise the rotten smell of
+careless neglect will eventually drive away contributors and/or co-workers.
+
+### Use Semicolons
+
+According to [scientific research][hnsemicolons], the usage of semicolons is
+a core value of our community. Consider the points of [the opposition][], but
+be a traditionalist when it comes to abusing error correction mechanisms for
+cheap syntactic pleasures.
+
+[the opposition]: http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding
+[hnsemicolons]: http://news.ycombinator.com/item?id=1547647
+
+### 80 characters per line
+
+Limit your lines to 80 characters. Yes, screens have gotten much bigger over the
+last few years, but your brain has not. Use the additional room for split screen,
+your editor supports that, right?
+
+### Use single quotes
```

Comment:
I would tend to disagree with this. We currently have ESLint set up to yell at us if we use single quotes, even if they are within another string.